lib/core/i32.carbon

```diff
@@ -0,0 +1,130 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
```

Comment:
Directory comment: maybe core/prelude/i32.carbon? You already have my `core/` comment on #naming, but I'll note here that I was also thinking maybe splitting off all the `prelude` files to a subdir (since we'll presumably have both libraries that users must import manually, and prelude libraries that are implicit)